app/Http/Controllers/NotificationController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Controller\ControllerPosition;
+use App\Models\Notification\Notification;
+use Illuminate\Http\JsonResponse;
+
+class NotificationController extends BaseController
+{
+    public function getActiveNotifications() : JsonResponse
+    {
+        $notifications = Notification::active()
+            ->orderBy('valid_from', 'desc')
+            ->with('controllers')
+            ->get()
+            ->each(function (Notification $notification) {
+
+                $notification->controllers->each(function (ControllerPosition $controllerPosition) {
+                    $controllerPosition->setHidden([
+                        'pivot', 'id', 'frequency', 'created_at', 'updated_at'
+                    ]);
+                });
+
+                $notification->setHidden([
+                    'created_at', 'updated_at', 'deleted_at'
+                ]);
+
+            });
+
+        return response()->json($notifications);
+    }
+
+    public function getUnreadNotifications() : JsonResponse
+    {
+        $notifications = Notification::active()
+            ->orderBy('valid_from', 'desc')
+            ->with('controllers')
+            ->doesntHave('readBy')
```

Comment:
This relationship wont discriminate based on who has read the notification, so will mark it read if any user has read it? Would it be good to create a local scope `scopeUnreadByUser` which makes sure it hasn't been read by the current user?

Reply:
67cdb63